### PR TITLE
Read GSM as client to figure out what map to load.

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
When a client connects, we now read the GSM to see what map to load rather than blindly loading the GameDefaultMap _then_ querying the GSM. 

#### Release note
Bugfix: Client's now load the map the server loads on connection.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -183,115 +160,157 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 		}
 	}
 
-	Connect();
+	Connection->Connect(bConnectAsClient);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: connect connect connect connect econnnt cecnoot

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 2]
```diff
@@ -424,9 +426,18 @@ void USpatialWorkerConnection::CacheWorkerAttributes()
 USpatialNetDriver* USpatialWorkerConnection::GetSpatialNetDriverChecked() const
 {
 	check(GEngine);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]:
```suggestion
```

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h [hunk 3]
```diff
@@ -37,7 +37,10 @@ class SPATIALGDK_API USpatialGameInstance : public UGameInstance
 private:
 
 	// TODO: Move SpatialConnection ownership to NetDriver
+	// TODO remove this
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h]: This is still needed right?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp [hunk 4]
```diff
@@ -424,9 +426,18 @@ void USpatialWorkerConnection::CacheWorkerAttributes()
...
+	UNetDriver* NetDriver = GameInstance->GetWorld()->GetNetDriver();
+
+	// On the client, the world might not be completely set up.
+	// in this case we can use the PendingNetGame to get the NetDriver
+	if (NetDriver == nullptr)
+	{
+		NetDriver = GameInstance->GetWorldContext()->PendingNetGame->GetNetDriver();
+	}
+
 	checkf(NetDriver, TEXT("SpatialNetDriver was invalid while accessing SpatialNetDriver!"));
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp]:
```suggestion
	checkf(Cast<USpatialNetDriver>(NetDriver), TEXT("SpatialNetDriver was invalid while accessing SpatialNetDriver!"));
```

File: SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h (hunk 3 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialGameInstance.h]: Eh we can just have an accessor. If other people want to use the connection that's fine.

**Pull request merged**